src/server.js

```diff
@@ -107,13 +107,10 @@ function setupRenderer() {
         console.error('ROUTER ERROR:', pretty.render(error));
         res.status(500);
         hydrateOnClient();
-      } else if (!renderProps) {
-        res.status(500);
-        hydrateOnClient();
-      } else {
+      } else if (renderProps) {
         const component = (
           <Provider store={store} key="provider">
-            <RoutingContext />
+            <RoutingContext {...renderProps} />
```

Comment:
Interesting.  Router 2.0 seems like a likely good idea generally, given the freshness of universal-redux (can we get a shorter name? my hands hurt :smile:).  Do you want me to fix this or do you want to close this and implement the changes independently?